source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -55,15 +55,36 @@ class ExtProcLoggingInfo : public Envoy::StreamInfo::FilterState::Object {
   explicit ExtProcLoggingInfo(const Envoy::ProtobufWkt::Struct& filter_metadata)
       : filter_metadata_(filter_metadata) {}
 
+  // gRPC call stats for headers and trailers.
   struct GrpcCall {
-    GrpcCall(const std::chrono::microseconds latency, const Grpc::Status::GrpcStatus status,
-             const ProcessorState::CallbackState callback_state)
-        : latency_(latency), status_(status), callback_state_(callback_state) {}
+    GrpcCall(const std::chrono::microseconds latency, const Grpc::Status::GrpcStatus call_status)
+        : latency_(latency), call_status_(call_status) {}
     const std::chrono::microseconds latency_;
-    const Grpc::Status::GrpcStatus status_;
-    const ProcessorState::CallbackState callback_state_;
```

Comment:
QQ: so we removed `ProcessorState::CallbackState` from new`GrpcCall` struct?

This field has been used widely (internally). It also provides the value to determine filter status. Can we keep it? It is just a enum so it should not bloat memory usage I think

Reply:
CallBackState basically is to tell the stats is for headers, trailers, or body.  With the latest data structure design which keeps separate header_stats, trailer_stats, body_stats, there is no need to keep that any more.

Reply:
From a quick look at how this field is being used in our code base, I feel it is more than just header, buffer, trailer etc. It also tell filter status. Also, It will requires a large amount of code changes for clients

Reply:
Based on my reading, CallbackState is only used for that purpose (identify it's header/trailer/body)for the stats parsing in client side.  Please ping me directly if you need more explanation on this change.

Overall, the requirement for the gRPC call stats is captured in this comment: https://github.com/envoyproxy/envoy/pull/27812#issuecomment-1594940341
Also, there are some comments stating that for body, only total_latency, max_latency, min_latency, call counts and last_call_status is interested.  So, the stats is designed this way.  Client should refactor based on this as well.  

Here are some examples:
grpcCalls(INBOUND).header_stats_ is the header stats for request.
grpcCalls(OUTBOUND).header_stats_ is the header stats for response.

grpcCalls(INBOUND).body_stats_ is the body stats for request.
grpcCalls(OUTBOUND).body_stats_ is the body stats for response.

grpcCalls(INBOUND).trailer_stats_ is the trailer stats for request.
grpcCalls(OUTBOUND).trailer_stats_ is the trailer stats for response.